extensions/2.0/Vendor/EXT_feature_metadata/README.md

```diff
@@ -0,0 +1,1691 @@
+
+* A scalar property cannot be encoded into multiple channels. For example, it is not possible to encode a `UINT32` property in an `RGBA8` texture.
+* Components of fixed-length array properties must be separate channels within the same texture.
+* Variable-length arrays are not supported.
+
+Additionally, the data type and bit depth of the image must be compatible with the property type. An 8-bit per pixel RGB image is only compatible with `UINT8` or normalized `UINT8` properties, and array properties thereof with three components or less. Likewise, a floating point property requires a floating point-compatible image format like KTX2 which may require additional extensions.
+
+Feature textures are defined with the following steps:
+
+1. A class is defined in the root `EXT_feature_metadata` extension object. This is used to describe the metadata in the texture.
+2. A feature texture is defined in the root `EXT_feature_metadata.featureTextures` object. This must reference the class ID defined in step 1.
+3. A feature texture is associated with a primitive by listing the feature texture ID in the `primitive.EXT_feature_metadata.featureTextures` array.
+<img src="figures/feature-texture.png"  alt="Feature Texture" width="500">
+
+_Class and feature texture_
+
+```jsonc
+{
+  "extensions": {
+    "EXT_feature_metadata": {
+      "schema": {
+        "classes": {
+          "heatSample": {
+            "properties": {
+              "heatSample": {
+                "type": "UINT8",
+                "normalized": true
+              }
+            }
+          }
+        }
+      },
+      "featureTextures": {
+        "heatLossTexture": {
+          "class": "heatSample",
+          "properties": {
+            "heatLoss": {
+              "texture": {
+                "index": 0,
+                "texCoord": 0
+              },
+              "channels": "r"
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+_Primitive_
+
+```jsonc
+{
+  "primitives": [
+    {
+      "attributes": {
+        "POSITION": 0,
+        "TEXCOORD_0": 1
+      },
+      "indices": 2,
+      "material": 0,
+      "extensions": {
+        "EXT_feature_metadata": {
+          "featureTextures": ["heatLossTexture"]
+        }
+      }
+    }
+  ]
+}
+```
+
+
+`texture` is a glTF [`textureInfo`](../../../../../specification/2.0/schema/textureInfo.schema.json) object. `texCoord` refers to the texture coordinate set of the referring primitive. `channels` is a string matching the pattern `"^[rgba]{1,4}$"` that specifies which texture channels store property values.
+
+### Statistics
+
+Statistics provide aggregate information about features in the model. Statistics are provided on a per-class basis.
+
+* `count` is the number of features that conform to the class
+* `properties` contains statistics about property values
```

Comment:
How critical are statistics to the initial version of 3D Tiles? I fully agree it's an important concept! But as "metadata about metadata" its scope is probably easier to separate, e.g. as `EXT_feature_metadata_statistics`, and a simpler scope might be a good thing for `EXT_feature_metadata` adoption.

Compared to the rest of this extension, I do have a number of somewhat open-ended concerns and suggestions about this section:

- What, if anything, do statistics mean for properties stored in a Feature Texture?
- Only allowing statistics on a per-class basis may be pretty limiting, especially outside of the 3D Tiles context, where a glTF asset represents an entire scene.
- Percentiles or quantiles are a common summary statistic; for something like showing a legend on a map visualization I think these would be very helpful.

Reply:
Another idea that came up was to remove statistics from the glTF extension but keep it in the 3D Tiles extension, where it feels more useful to have aggregate information.

The question about feature textures still remains. Conceptually each texel in the feature texture that's visible on the 3D surface would be considered a feature. That's a bit awkward to describe though.

:+1: on percentiles or quantiles

Reply:
yeah defining things for feature textures is tricky, you would want to sum/average over only valid texels, though that requires a definition of what "valid" means.

From past experience in applications that do this, you need some sort of sentinel value for "no data". And I've seen that vary a lot.  e.g. the GTX grid shift file format uses an oddly specific value of `-88.88880` https://vdatum.noaa.gov/docs/gtx_info.html for "no data".

But in the call yesterday, @donmccurdy brought up a good point, if you were computing an average, you might not want the raw average over the number of texels, you might want to weight by the area the texture takes up in the world.

Reply:
Proposed removing statistics from this extension (with possible addition as a later extension), in https://github.com/CesiumGS/glTF/pull/13.